src/util/ParserUtil.h

```diff
@@ -0,0 +1,75 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#ifndef UTIL_PARSERUTIL_H_
+#define UTIL_PARSERUTIL_H_
+
+#include "common/base/Base.h"
+#include "common/base/StatusOr.h"
+#include "context/QueryContext.h"
+#include "parser/MaintainSentences.h"
+#include "visitor/RewriteMatchLabelVisitor.h"
+
+namespace nebula {
+namespace graph {
+
+class ParserUtil final {
+public:
+    ParserUtil() = delete;
+
+    static bool isLabel(const Expression *expr) {
+        return expr->kind() == Expression::Kind::kLabel ||
+               expr->kind() == Expression::Kind::kLabelAttribute;
+    }
+
+    static void rewritePred(QueryContext *qctx,
+                          PredicateExpression *pred,
+                          const std::string &oldVarName) {
+        const auto &newVarName = qctx->vctx()->anonVarGen()->getVar();
+        qctx->ectx()->setValue(newVarName, Value());
+        auto rewriter = [&oldVarName, &newVarName](const Expression *expr) {
+            Expression *ret = nullptr;
+            if (expr->kind() == Expression::Kind::kLabel) {
+                auto *label = static_cast<const LabelExpression *>(expr);
+                if (*label->name() == oldVarName) {
+                    ret = new VariableExpression(new std::string(newVarName));
+                } else {
+                    ret = label->clone().release();
+                }
+            } else {
+                DCHECK(expr->kind() == Expression::Kind::kLabelAttribute);
+                auto *la = static_cast<const LabelAttributeExpression *>(expr);
+                if (*la->left()->name() == oldVarName) {
+                    const auto &value = la->right()->value();
+                    ret =
+                        new AttributeExpression(new VariableExpression(new std::string(newVarName)),
+                                                new ConstantExpression(value));
+                } else {
+                    ret = la->clone().release();
+                }
+            }
+            return ret;
+        };
+
+        RewriteMatchLabelVisitor visitor(rewriter);
+
+        pred->setOriginString(new std::string(pred->makeString()));
+        pred->setInnerVar(new std::string(newVarName));
```

Comment:
I think we should not simply replace the original var name, or `pred ->toString()` will not give a proper string of the PredicateExpression.

Reply:
I have use method `setOriginString()` to save the origin string of predicate expression before it is rewritten, so `pred->toString()` will show the origin string properly:
```
std::string PredicateExpression::toString() const {
    if (originString_ != nullptr) {
        return *originString_;
    } else {
        return makeString();
    }
}